src/com/textFinder/TextFinderImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.textFinder;
+
+import java.io.File;
+import java.util.*;
+import java.util.concurrent.*;
+
+import com.textFinder.finder.FileSearchThreadRunner;
+
+public class TextFinderImpl implements TextFinder {
+
+    @Override
+    public List<File> findText(File file, String searchText) {
+        File absoluteFile = file.getAbsoluteFile();
+
+        List<File> result = Collections.synchronizedList(new ArrayList<>());
+        ExecutorService executor = Executors.newCachedThreadPool();
+        List<Future<?>> futures = new ArrayList<>();
```

Comment:
`Map<String, Future<Boolean>>`: File path vs whether this future of "file contains the search text or not"